pkg/sql/sem/tree/casts.go

```diff
@@ -808,6 +808,11 @@ func performCastWithoutPrecisionTruncation(ctx *EvalContext, d Datum, t *types.T
 		}
 
 	case types.StringFamily, types.CollatedStringFamily:
+		if w, ok := d.(*DOidWrapper); ok {
+			// If we're casting a DOidWrapper, then we want to cast the wrapped
+			// datum. It is also reasonable to lose the old Oid value too.
+			d = w.Wrapped
```

Comment:
I think you want to use the `UnwrapDatum` method here, which will unwrap a whole stack of DOidWrappers if there is one.

Reply:
Then again, we should probably just eagerly do that up top.